base/server/healthcheck/pki/server/healthcheck/meta/csconfig.py

```diff
@@ -56,6 +56,13 @@ def compare_nssdb_with_cs(class_instance, subsystem, cert_tag):
                           nssdbDir=subsystem.instance.nssdb_dir,
                           msg='Unable to load cert from NSSDB: %s' % str(e))
 
+    if cert_nssdb == None:
+        logger.debug('No cert in NSSDB with nickname: %s', cert['nickname'])
+        return Result(class_instance, constants.ERROR,
+                      key=cert_id,
+                      nssdbDir=subsystem.instance.nssdb_dir,
+                      msg='No cert in NSSDB with nickname: %s' % cert['nickname'])
```

Comment:
We probably should include the token name (if available) since IPA is planning to support HSM, so it should be just `<nickname>` if there's no token, or `<token>:<nickname>` if there's a token.

Reply:
Good idea, I added the token name in conditionally if it exists.